src/mono/netcore/System.Private.CoreLib/src/System/Environment.iOS.cs

```diff
@@ -94,6 +93,14 @@ private static string GetFolderPathCore(SpecialFolder folder, SpecialFolderOptio
                 default:
                     return string.Empty;
             }
+
+            static string? CombineSearchPath(NSSearchPathDirectory searchPath, string subdirectory)
+            {
+                string? path = Interop.Sys.SearchPath(searchPath);
+                return path != null ?
+                    Path.Combine(path, subdirectory) :
+                    null;
```

Comment:
> it lets us simplify the nullability flow

Changed, but it doesn't let us simplify any of the call sites currently.  As written, GetSpecialFolder still needs to be string? because SearchPath is string? and its result is directly returned.